crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1990,3 +2244,37 @@ where
             .tap_err(|e| info!(?digest, "execute_cert_to_true_effects failed: {}", e))
     }
 }
+
+/// Given an AuthorityAggregator on genesis (epoch 0), catch up to the latest epoch and fill in
+/// all past epoches' committee information.
+pub async fn reconfig_from_genesis(
+    mut aggregator: AuthorityAggregator<NetworkAuthorityClient>,
+) -> SuiResult<AuthorityAggregator<NetworkAuthorityClient>> {
+    fp_ensure!(
+        aggregator.committee.epoch == 0,
+        SuiError::from("reconfig_from_genesis entails an authority aggregator with epoch 0")
+    );
+    let latest_committee = aggregator.get_committee_with_net_addresses(0).await?;
+    let latest_epoch = latest_committee.committee.epoch;
+    if latest_epoch == 0 {
+        // If still at epoch 0, no need to reconfig
+        return Ok(aggregator);
+    }
+    // First we fill in the committee store from 1 to latest_epoch - 1
+    let mut cur_epoch = 1;
+    let network_config = default_mysten_network_config();
+    loop {
+        if cur_epoch >= latest_epoch {
+            break;
+        }
+        let committee = Committee::try_from(aggregator.get_committee_info(Some(cur_epoch)).await?)?;
```

Comment:
Is the quorum created from genesis configuration guaranteed to know about future epochs? If we removed validators since genesis, could this still fail?

Reply:
no if most of the genesis committee members stopped serving then this won't work. See "**Caveat**" in this PR's description